saleor/tests/e2e/checkout/test_checkout_calculate_correct_discount_for_sale_and_voucher.py

```diff
@@ -0,0 +1,181 @@
+import pytest
+
+from ..product.utils.preparing_product import prepare_product
+from ..sales.utils import create_sale, create_sale_channel_listing, sale_catalogues_add
+from ..shop.utils.preparing_shop import prepare_shop
+from ..utils import assign_permissions
+from ..vouchers.utils import create_voucher, create_voucher_channel_listing
+from .utils import (
+    checkout_add_promo_code,
+    checkout_complete,
+    checkout_create,
+    checkout_delivery_method_update,
+    checkout_dummy_payment_create,
+    checkout_lines_add,
+)
+
+
+def prepare_sale_for_variant(e2e_staff_api_client, channel_id, product_variant_id):
```

Comment:
Maybe as an input you could have optional `discount_type`?
```python
def prepare_sale_for_variant(e2e_staff_api_client, channel_id, product_variant_id, sale_discount_type="PERCENTAGE"):
    ...
```
Then your function, can prepare different type of vouchers